### PR TITLE
feat: make hover evaluation timeout configurable

### LINKED ISSUE #1242: Make timeout for evaluation on hover configurable
**Is your feature request related to a problem? Please describe.**
Currently, value evaluation for hovering [is timed out after 500ms](https://github.com/microsoft/vscode-js-debug/blob/1d734dea33cd4de9c85f65a862d3daff26ea0f81/src/adapter/threads.ts#L438). If CDP communication plus evaluation within the browser takes longer than 500ms, no hover value is shown.

**Describe the feature you'd like**
The situation can be resolved in one of the following ways.

- Increase the timeout
- Remove the timeout all together
- Make the timeout configurable within launch or workspace options

Furthermore, VS Code hovering delays the creation of the debug hover by one second to make the UI less hectic (see [#2309](https://github.com/Microsoft/vscode/issues/2309)). Evaluation could be started as soon as hovering is detected, so that the time required for evaluation falls within the UI delay. But this would likely require changes in VS Code itself.

### PR DESCRIPTION
**Pull request by timvahlbrock**
Fixes #1242

### CODE REVIEW

File: src/configuration.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export interface IBaseConfiguration extends IMandatedConfiguration {
   /**
    * Timeouts for several operations (currently only source-maps)
    */
-  timeouts: Partial<SourceMapTimeouts>;
+  timeouts: Partial<SourceMapTimeouts> & HoverEvaluationTimeout;
```
Comment (connor4312) [inline src/configuration.ts:149]: I think we can instead rename `SourceMapTimeouts` and add `hoverEvaluation` to it

Comment (timvahlbrock) [reply src/configuration.ts:149]: Thought about this already, but I wasn't sure, whether I should move the definition of `SourceMapTimeouts` then.

Comment (timvahlbrock) [reply src/configuration.ts:149]: The `SourceMapTimeouts` type is used in the `Sources` class. If the type is modified & renamed, anywhere in `Sources` where `SourceMapTimeouts` is currently used, the `hoverEvaluation` timeout needs to have been set, although `Sources` although `Sources` is completely unrelated to it and therefor never uses it.

Comment (timvahlbrock) [reply src/configuration.ts]: I pushed a commit where `SourceMapTimeouts` still is a separate interface, but the types are structured a little bit better.

File: src/adapter/threads.ts [hunk 2]
```diff
@@ -497,6 +497,17 @@ export class Thread implements IVariableStoreLocationProvider {
     };
   }
 
+  private getHoverEvalTimeout() {
+    const configuredTimeout = this.launchConfig.timeouts?.hoverEvaluation;
+    if (configuredTimeout === undefined) {
+      return 10 * 1000;
```
Comment (connor4312) [inline src/adapter/threads.ts:503]: Btw, TS (and recent JS) allows separators so you can write this as `10_000`

**Review by connor4312 — APPROVED** [review]
Lgtm, thanks!

**Pull request merged**